qgym/benchmarks/metrics/initial_mapping_metrics.py

```diff
@@ -48,3 +71,72 @@ def __init__(
 
         Args:
         """
+
+
+class InitialMappingBenchmarker:
+
+    def __init__(
+        self,
+        generator: GraphGenerator | None = None,
+        *,
+        metrics: Iterable[InitialMappingMetric] | InitialMappingMetric,
+    ) -> None:
+        """Init of the :clas:`InitialMappingBenchmarker` class.
+
+        Args:
+            generator: Interaction graph generator to use during benchmarking
+            metrics: Metrics to compute.
+        """
+        self.generator = BasicGraphGenerator() if generator is None else generator
+        self.metrics = (
+            (metrics,) if isinstance(metrics, InitialMappingMetric) else tuple(metrics)
+        )
+        for metric in self.metrics:
+            if hasattr(metric, "connection_graph"):
+                connection_graph = deepcopy(metric.connection_graph)
+                self.generator.set_state_attributes(connection_graph=connection_graph)
+                break
+
+    def run(
+        self,
+        mapper: Mapper,
+        max_iter: int = 1000,
+        return_type: str = "raw",
+    ) -> NDArray[np.float64]:
+        """Run the benchmark.
+
+        Args:
+            mapper: Mapper to benchmark.
+            max_iter: Maximum number of iterations to benchmark.
+            return_type: Return type to return the results. Choose from ["raw",
+                "quartiles", "median", "mean"]
+
+        Returns:
+            NDArray containing the results from the benchmark.
+        """
+        return_type = check_string(return_type, "return_type", lower=True)
+
+        results: list[deque[float]] = [deque() for _ in self.metrics]
+        for i, interaction_graph in enumerate(self.generator, start=1):
+            mapping = mapper.compute_mapping(interaction_graph)
+            for metric, result_que in zip(self.metrics, results):
+                result_que.append(metric.compute(interaction_graph, mapping))
+
+            if i >= max_iter:
+                break
+
+        if return_type == "raw":
+            return cast(NDArray[np.float64], np.array(results, dtype=np.float64))
+        if return_type == "quartiles":
+            return cast(
+                NDArray[np.float64],
+                np.quantile(results, [0, 0.25, 0.5, 0.75, 1], axis=1),
+            )
+        if return_type == "median":
+            return cast(NDArray[np.float64], np.median(results, axis=1))
+        if return_type == "mean":
+            return cast(NDArray[np.float64], np.mean(results, axis=1))
+
+        raise ValueError(
+            "Unknown 'return_type'. Choose from ['raw', 'quartiles', 'median', 'mean']"
+        )
```

Comment:
Maybe put this class into another file, because the benchmarker is not a metric (which the file title suggest).